tests/Storage/ObjectTest.php

```diff
@@ -72,6 +73,73 @@ public function testUpdatesData()
         $this->assertEquals($data['contentType'], $object->info()['contentType']);
     }
 
+    public function testCopyObjectWithDefaultName()
+    {
+        $sourceBucket = 'bucket';
+        $destinationBucket = 'bucket2';
+        $objectName = 'object.txt';
+        $acl = 'private';
+        $this->connection->copyObject([
+                'sourceBucket' => $sourceBucket,
+                'sourceObject' => $objectName,
+                'destinationBucket' => $destinationBucket,
+                'destinationObject' => $objectName,
+                'destinationPredefinedAcl' => $acl
+            ])
+            ->willReturn([
+                'bucket' => $destinationBucket,
+                'name' => $objectName,
+                'generation' => 1
+            ]);
+        $object = new Object($this->connection->reveal(), $objectName, $sourceBucket);
+        $copiedObject = $object->copy($destinationBucket, [
+            'predefinedAcl' => $acl
+        ]);
+
+        $this->assertEquals($destinationBucket, $copiedObject->info()['bucket']);
+        $this->assertEquals($objectName, $copiedObject->info()['name']);
+    }
+
+    public function testCopyObjectWithNewName()
+    {
+        $sourceBucket = 'bucket';
+        $sourceObject = 'object.txt';
+        $bucketConnection = $this->prophesize('Google\Cloud\Storage\Connection\ConnectionInterface')->reveal();
+        $destinationBucketName = 'bucket2';
+        $destinationBucket = new Bucket($bucketConnection, $destinationBucketName);
+        $destinationObject = 'object2.txt';
+        $acl = 'private';
+        $this->connection->copyObject([
+                'sourceBucket' => $sourceBucket,
+                'sourceObject' => $sourceObject,
+                'destinationBucket' => $destinationBucketName,
+                'destinationObject' => $destinationObject,
+                'destinationPredefinedAcl' => $acl
+            ])
+            ->willReturn([
+                'bucket' => $destinationBucketName,
+                'name' => $destinationObject,
+                'generation' => 1
+            ]);
+        $object = new Object($this->connection->reveal(), $sourceObject, $sourceBucket);
+        $copiedObject = $object->copy($destinationBucket, [
+            'predefinedAcl' => $acl,
+            'name' => $destinationObject
+        ]);
+
+        $this->assertEquals($destinationBucketName, $copiedObject->info()['bucket']);
+        $this->assertEquals($destinationObject, $copiedObject->info()['name']);
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testCopyObjectThrowsException()
```

Comment:
Nitpick: Could you change this to `testCopyObjectInvalidDestinationType` or something that tells why it is erroring?